packages/manager/modules/pci/src/projects/project/instances/add/add.controller.js

```diff
@@ -314,14 +379,271 @@ export default class PciInstancesAddController {
+      `instances_create_${
+        this.selectedCategory
+      }_${mode}_${this.instance.region.toLowerCase()}_${
+        this.model.flavorGroup.name
+      }_${this.selectedPrivateNetwork.name}`,
+    );
+  }
+
+  onModeFocus() {
+    this.displaySelectedMode = false;
+  }
+
+  onModeChange() {
+    this.displaySelectedMode = true;
+  }
+
+  onAttachIPChange(value) {
+    if (value && !this.floatingIps) {
+      this.getFloatingIps();
+    }
+    if (value && this.isCreateFloatingIPClicked) {
+      this.isCreateFloatingIPClicked = false;
+    }
+    if (value) {
+      this.trackInstance(`add::enable-attach-floating-ip`);
+    } else {
+      this.trackInstance(`add::disable-attach-floating-ip`);
+    }
+    return this.floatingIps;
+  }
+
+  onSelectedModeChange(mode) {
+    this.selectedMode = mode;
+    this.selectedPrivateNetwork = {
+      id: '',
+      name: this.$translate.instant(
+        'pci_projects_project_instances_add_privateNetwork_none',
+      ),
+    };
+    if (this.subnetGateways) this.subnetGateways.length = 0;
+    if (this.isPrivateMode()) {
+      this.isCreateFloatingIPClicked = false;
+      this.isAttachFloatingIPAvailable = false;
+      this.selectedPrivateNetwork = {
+        id: '',
+        name: this.$translate.instant(
+          'pci_projects_project_instances_add_privateNetwork_placeholder',
+        ),
+      };
+    }
+  }
+
+  isPrivateMode() {
+    return this.selectedMode.name === this.instanceModeEnum[1].mode;
+  }
+
+  onFloatingIpChange(value) {
+    if (value !== null && !value?.id) {
+      this.isCreateFloatingIPClicked = true;
+      this.trackInstance(`add::create-floating-ip`);
+    } else {
+      this.selectedFloatingIP = value;
+      this.isCreateFloatingIPClicked = false;
+    }
+  }
+
+  getPrivateNetworkOnChange() {
+    if (get(this.selectedPrivateNetwork, 'id')) {
+      this.instance.networks = [
+        {
+          networkId: get(this.selectedPrivateNetwork, 'id'),
+        },
+      ];
+    } else {
+      this.instance.networks = [];
+    }
+  }
+
+  onPrivateNetworkChange(modelValue) {
+    this.getPrivateNetworkOnChange();
+    if (
+      modelValue &&
+      modelValue.subnet &&
+      this.selectedMode.name !== this.instanceModeEnum[1].mode
+    ) {
+      this.isGatewayLoading = true;
+      this.getSubnetGateways(modelValue.subnet[0].id).then((data) => {
+        this.subnetGateways = data;
+      });
+    }
+    this.subnetGateways = [];
+  }
+
+  isNetworkExists() {
+    return this.availablePrivateNetworks.some(
+      (network) => network.name === this.privateNetworkModel.name,
+    );
+  }
+
+  onAddPrivateNetworkClick() {
+    this.trackInstance(`add::create-private-network`);
+    this.showAddPrivateNetworkModalForm = true;
+    if (this.showAddPrivateNetworkModalForm) {
+      this.trackInstance(`add::create-private-network`);
+    }
+    const randomNumber = Math.floor((1 + Math.random()) * 0x10000)
+      .toString(16)
+      .substring(1);
```

Comment:
question, did you try to look if we have a module or NPM package to generate Random AlphaNum values ?

Reply:
@mohammed-zahaf  yes we have a module, I updated this using that module.

Reply:
Cool 👍